src/compute-client/src/controller.rs

```diff
@@ -515,69 +524,21 @@ where
     ComputeGrpcClient: ComputeClient<T>,
 {
     /// Adds replicas of an instance.
-    pub async fn add_replica_to_instance(
+    pub fn add_replica_to_instance(
         &mut self,
         instance_id: ComputeInstanceId,
         replica_id: ReplicaId,
         config: ComputeReplicaConfig,
     ) -> Result<(), ComputeError> {
-        let (addrs, communication_config) = match config.location {
-            ComputeReplicaLocation::Remote {
-                addrs,
-                compute_addrs,
-                workers,
-            } => {
-                let addrs = addrs.into_iter().collect();
-                let comm = CommunicationConfig {
-                    workers: workers.get(),
-                    process: 0,
-                    addresses: compute_addrs.into_iter().collect(),
-                };
-                (addrs, comm)
-            }
-            ComputeReplicaLocation::Managed {
-                allocation,
-                availability_zone,
-                ..
-            } => {
-                let service = self
-                    .compute
-                    .orchestrator
-                    .ensure_replica(instance_id, replica_id, allocation, availability_zone)
-                    .await?;
-
-                let addrs = service.addresses("controller");
-                let comm = CommunicationConfig {
-                    workers: allocation.workers.get(),
-                    process: 0,
-                    addresses: service.addresses("compute"),
-                };
-                (addrs, comm)
-            }
-        };
-
-        self.instance(instance_id)?.add_replica(
-            replica_id,
-            addrs,
-            config.logging,
-            communication_config,
-        )
+        self.instance(instance_id)?.add_replica(replica_id, config)
```

Comment:
What's the state of the replica after this returns? Are any introspection queries that a users runs incorrect during the time when it's provisioning before it's ready? (Does ready even mean anything? I don't know.) Is "not yet ready" the same as "ready but not up to date yet" so it's the same to the user?

Reply:
It should be the same as before (apart from the orchestrator state). 

Before: after add_replica has been `.await`ed we were sure that the per replica catalog entries (introspection and system tables) are present in the catalog and the replica was registered with the orchestrator. But no guarantee about the replica being actually up or hydrated.

Now: After `add_replica` has finished (not async anymore) we can be sure that the per replica catalog entries are present. No guarantee anymore about the orchestrator state, but the background task will `ensure` this in the orchestrator until successful.